nima_core/storage/persistent_memory.py

```
@@ -150,7 +150,13 @@ def secure_torch_load(filepath: Path, verify: bool = True, map_location=None) ->
     # Load with torch
     if map_location is None:
         map_location = 'cpu'
-    return torch.load(filepath, map_location=map_location, weights_only=False)
+    # SECURITY: weights_only=True prevents RCE via malicious pickle
+    try:
+        return torch.load(filepath, map_location=map_location, weights_only=True)
+    except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)